spec/index.bs

```diff
@@ -2564,6 +2565,80 @@ If a format is listed in [[#plain-color-formats]] with {{GPUTextureUsage/RENDER_
 color renderable format. Any other format is not a color renderable format. Any depth/stencil format is a
 depth or stencil renderable format. Any other format is not a depth or stencil renderable format.
 
+## <dfn interface>GPUExternalTexture</dfn> ## {#gpu-external-texture}
+
+A {{GPUExternalTexture}} is a sampleable texture wrapping an external canvas or video object.
+The contents of a {{GPUExternalTexture}} object may not change, either from inside WebGPU
+(it is only sampleable) or from outside WebGPU (e.g. canvas drawing or video frame advancement).
+
+Note: External textures *can* be implemented without creating a copy of the imported canvas/video,
+but a user agent may not always implement this for all cases.
+
+<script type=idl>
+[Exposed=Window]
+interface GPUExternalTexture {
+};
+GPUExternalTexture includes GPUObjectBase;
+</script>
+
+{{GPUExternalTexture}} has the following internal slots:
+
+<dl dfn-type=attribute dfn-for="GPUExternalTexture">
+    : <dfn>\[[destroyed]]</dfn>, of type `boolean`
+    ::
+        Indicates whether the object has been destroyed (can no longer be used).
+        Initially set to `false`.
+
+        <!-- Move this to an internal object if GPUExternalTexture becomes Serializable. -->
+</dl>
+
+### Importing External Textures ### {#external-texture-creation}
+
+An external texture is created from an external video object
+using {{GPUDevice/importExternalTexture()}}.
+
+External textures are destroyed automatically, as a microtask,
+instead of manually or upon garbage collection like other resources.
+
+<script type=idl>
+dictionary GPUExternalTextureDescriptor : GPUObjectDescriptorBase {
+    required HTMLVideoElement source;
+};
+</script>
+
+<dl dfn-type=method dfn-for=GPUDevice>
+    : <dfn>importExternalTexture(descriptor)</dfn>
+    ::
+        Creates a {{GPUExternalTexture}} wrapping the provided image source.
+
+        <div algorithm=GPUDevice.importExternalTexture>
+            **Called on:** {{GPUDevice}} this.
+
+            **Arguments:**
+            <pre class=argumentdef for="GPUDevice/importExternalTexture(descriptor)">
+                |descriptor|: Provides the external image source object (and any creation options).
+            </pre>
+
+            **Returns:** {{GPUExternalTexture}}
+
+            1. Let |source| be |descriptor|.{{GPUExternalTextureDescriptor/source}}.
+
+            1. If |source| <l spec=html>[=is not origin-clean=]</l>,
+                throw a {{SecurityError}} and stop.
+
+            1. Let |result| be a new {{GPUExternalTexture}} object wrapping a copy of the current
+                image contents of |source|.
+
+                Note: This is described as a copy, but may be implemented without an actual copy.
+
+            1. [=Queue a microtask=] to set |result|.{{GPUExternalTexture/[[destroyed]]}} to `true`,
+                releasing the underlying resource.
+
+                Issue: Is this too restrictive?
```

Comment:
Maybe? We should probably discuss it in the group as a follow-up. An option would be to add a `retained: boolean` to the descriptor that would make that micro-task not enqueued but could have additional cost (like an extra copy to decouple the lifetime of the texture).